cmd/keeper/keeper.go

```diff
@@ -451,6 +452,14 @@ func (p *PostgresKeeper) GetPGState(pctx context.Context) (*cluster.PostgresStat
 		}
 		pgState.ReplicationLag = replicationLag
 
+		ctx, cancel = context.WithTimeout(pctx, p.clusterConfig.RequestTimeout)
+		role, err := pg.GetRole(ctx, p.getLocalConnParams())
+		defer cancel()
+		if err != nil {
+			return nil, fmt.Errorf("error getting replication lag: %v", err)
```

Comment:
can you start using package `trace` instead of built-in `fmt` as we need to track origins of error.